src/Core/Services/MetadataProviders/ISqlMetadataProvider.cs

```diff
@@ -72,6 +72,13 @@ public interface ISqlMetadataProvider
         // of referencing and referenced tables.
         public Dictionary<RelationShipPair, ForeignKeyDefinition>? PairToFkDefinition { get; }
 
+        /// <summary>
+        /// Maps {entityName, relationshipName} to the ForeignKeyDefinition defined for the relationship.
```

Comment:
might need to update the comment to indicate that this only stores relevant data for self-referencing relationships.